jf_agent/git/github.py

```diff
@@ -13,6 +13,85 @@
 _repo_redactor = NameRedactor()
 
 
+@diagnostics.capture_timing()
+@agent_logging.log_entry_exit(logger)
+def load_and_dump(config, endpoint_git_instance_info, git_conn):
+    write_file(
+        config.outdir,
+        'bb_users',
+        config.compress_output_files,
+        get_users(git_conn, config.git_include_projects),
+    )
+
+    write_file(
+        config.outdir,
+        'bb_projects',
+        config.compress_output_files,
+        get_projects(git_conn, config.git_include_projects, config.git_redact_names_and_urls),
+    )
+
+    api_repos = None
+
+    @diagnostics.capture_timing()
+    @agent_logging.log_entry_exit(logger)
+    def get_and_write_repos():
+        nonlocal api_repos
```

Comment:
nit: this strikes me as a code smell, any reason not to pull this out into a method like `get_projects` or `get_users`?

Reply:
Right. I pulled this directly from the implementation for BBS, and GitHub. I'm not really sure of the purpose, but ended up keeping it in the refactor.